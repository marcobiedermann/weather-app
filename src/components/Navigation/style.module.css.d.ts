declare const styles: {
  readonly navigation: string;
  readonly navigation__list: string;
  readonly navigation__item: string;
  readonly navigation__link: string;
};
export = styles;
