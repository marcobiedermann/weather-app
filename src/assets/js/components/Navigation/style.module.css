/* stylelint-disable block-no-empty */

.navigation {
  text-transform: uppercase;
}

.navigation__list {
  list-style: none;
  margin-bottom: 1.5em;
  margin-top: 0;
  padding: 0;
}

.navigation__item {
}

.navigation__link {
  padding-bottom: 0.25em;
  padding-top: 0.25em;
}
