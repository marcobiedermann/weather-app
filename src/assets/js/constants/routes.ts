export const INDEX = {
  name: 'Home',
  path: '/',
};

export const CITIES = {
  name: 'Cities',
  path: '/cities',
};

export const SEARCH = {
  name: 'Search',
  path: '/search',
};

export const SETTINGS = {
  name: 'Settings',
  path: '/settings',
};
