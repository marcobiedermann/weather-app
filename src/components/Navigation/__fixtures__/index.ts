const route = {
  name: 'Home',
  path: '/',
};

const routes = [route];

export default routes;
