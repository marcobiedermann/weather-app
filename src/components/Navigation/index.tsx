import React, { FC } from 'react';
import { Link } from 'react-router-dom';
import styles from './style.module.css';

interface Route {
  name: string;
  path: string;
}

export interface NavigationProps {
  routes: Route[];
}

const Navigation: FC<NavigationProps> = (props) => {
  const { routes, ...otherProps } = props;

  return (
    <nav className={styles.navigation} {...otherProps}>
      <ul className={styles.navigation__list}>
        {routes.map((route) => (
          <li key={route.path} className={styles.navigation__item}>
            <Link to={route.path} className={styles.navigation__link}>
              {route.name}
            </Link>
          </li>
        ))}
      </ul>
    </nav>
  );
};

export default Navigation;
