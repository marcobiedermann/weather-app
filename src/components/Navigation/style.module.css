/* stylelint-disable block-no-empty */

.navigation {
}

.navigation__list {
  list-style: none;
  margin: 0;
  padding: 0;
}

.navigation__item {
}

.navigation__link {
  align-items: center;
  display: flex;
  gap: 0.5em;
  padding: 0.5em 3em;
  position: relative;
  text-decoration: none;
}

.navigation__linkActive::after {
  background-color: currentcolor;
  content: '';
  height: 100%;
  position: absolute;
  right: 0;
  width: 3px
}