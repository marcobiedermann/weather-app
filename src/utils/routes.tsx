import { lazy } from 'react';
import { createBrowserRouter, Navigate } from 'react-router';
import Layout from '../components/Layout';

const CitiesPage = lazy(() => import('../pages/Cities'));
const CityPage = lazy(() => import('../pages/Cities/City'));
const CitiesEditPage = lazy(() => import('../pages/Cities/Edit'));
const SearchPage = lazy(() => import('../pages/Search'));
const SettingsPage = lazy(() => import('../pages/Settings'));

const routes = [
  {
    element: <Layout />,
    children: [
      {
        element: <Navigate to="/cities" />,
        path: '/',
      },
      {
        element: <CitiesPage />,
        path: '/cities',
      },
      {
        element: <CitiesEditPage />,
        path: '/cities/edit',
      },
      {
        element: <CityPage />,
        path: '/cities/:cityId',
      },
      {
        element: <SearchPage />,
        path: '/search',
      },
      {
        element: <SettingsPage />,
        path: '/settings',
      },
    ],
  },
];

const router = createBrowserRouter(routes);

export { router, routes };
