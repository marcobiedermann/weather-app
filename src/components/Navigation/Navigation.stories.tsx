import type { Meta, StoryObj } from '@storybook/react';
import { MemoryRouter } from 'react-router-dom';
import Navigation from '.';

const meta: Meta<typeof Navigation> = {
  component: Navigation,
  args: {
    routes: [
      {
        name: 'Home',
        path: '/',
      },
    ],
  },
  decorators: [
    (Story) => (
      <MemoryRouter>
        <Story />
      </MemoryRouter>
    ),
  ],
};

type Story = StoryObj<typeof Navigation>;

const Default: Story = {};

export { Default };
export default meta;
