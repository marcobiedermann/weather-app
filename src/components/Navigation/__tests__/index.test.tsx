import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { describe, expect, it } from 'vitest';
import Navigation from '..';
import routes from '../__fixtures__';

describe('components/Navigation', () => {
  it('should render correctly', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Navigation routes={routes} />
      </MemoryRouter>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render empty routes correctly', () => {
    const { asFragment } = render(<Navigation routes={[]} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
