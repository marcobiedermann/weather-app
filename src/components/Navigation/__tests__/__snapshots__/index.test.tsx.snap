// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Navigation renders correctly 1`] = `
<DocumentFragment>
  <nav
    class="navigation"
  >
    <ul
      class="navigation__list"
    />
  </nav>
</DocumentFragment>
`;
